go/externals/proof_support_facebook.go

```diff
@@ -95,36 +97,41 @@ func (rc *FacebookChecker) CheckStatus(ctx libkb.ProofContext, h libkb.SigHint)
 }
 
 func (rc *FacebookChecker) CheckStatusOld(ctx libkb.ProofContext, h libkb.SigHint) libkb.ProofError {
-	// Previously we would parse markup to extract the username and proof text.
-	// We had to switch to the desktop site to validate the usernames of people
-	// with the "no search engine scraping" Facebook privacy setting turned on.
-	// (Since only the desktop site, and not the m-site, enforces that the
-	// author ID in a post URL is correct.) That in turn made it much harder to
-	// parse the markup we get, because what we want comes down in an embedded
-	// comment. However, because the desktop site (again, unlike the m-site)
-	// does not display comments or ads to logged-out users, we can do a simple
-	// string match on the whole page to find the proof text. It's possible
-	// there's some way I haven't thought of for other users to inject strings
-	// somewhere in this page, or break that URL->author relationship, and if
-	// so we'll need to change it.
-
 	desktopURL := makeDesktopURL(h.GetAPIURL())
 
-	res, err := ctx.GetExternalAPI().GetText(libkb.NewAPIArg(desktopURL))
+	res, err := ctx.GetExternalAPI().GetHTML(libkb.NewAPIArg(desktopURL))
 	if err != nil {
 		return libkb.XapiError(err, desktopURL)
 	}
 
-	// We require a tiny bit of structure, which is that the proof must appear
-	// as the exact contents of an <a> tag. Again, this is just a textual
-	// search -- the <a> tag in question is actually expected to be in a
-	// comment.
-	escapedProofText := regexp.QuoteMeta(h.GetCheckText())
-	proofTextRegex := regexp.MustCompile("<a[^>]*>\\s*" + escapedProofText + "\\s*</a[^>]*>")
+	// Get the contents of the first (only) comment inside the first <code>
+	// block. Believe it or not, this comment contains the post markup below.
+	firstCodeCommentElements := res.GoQuery.Find("code").Eq(0).Contents().Nodes
+	if len(firstCodeCommentElements) == 0 {
+		return libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "failed to find proof markup comment in Facebook response")
+	}
+	firstCodeComment := firstCodeCommentElements[0].Data
+
+	// Facebook escapes "--" as "-\-\" and "\" as "\\" when inserting text into
+	// comments. Unescape these. (Use split-join instead of replace to get all
+	// occurrences without worrying about regex metacharacters.)
+	unescapedComment := strings.Replace(firstCodeComment, "-\\-\\", "--", -1)
+	unescapedComment = strings.Replace(unescapedComment, "\\\\", "\\", -1)
```

Comment:
The comment says use "split-join" instead of "replace" but then the code uses `strings.Replace`.